modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueArea.scala

```diff
@@ -3,152 +3,192 @@
+import edu.gemini.seqexec.model.Model.{Instrument, SequenceId, DaytimeCalibrationTargetName, SequenceState}
 import edu.gemini.seqexec.web.client.circuit._
 import edu.gemini.seqexec.web.client.actions._
 import edu.gemini.seqexec.web.client.model.Pages._
 import edu.gemini.seqexec.web.client.ModelOps._
+import edu.gemini.seqexec.web.client.services.HtmlConstants.iconEmpty
 import edu.gemini.seqexec.web.client.semanticui.elements.icon.Icon.{IconAttention, IconCheckmark, IconCircleNotched, IconSelectedRadio}
-import edu.gemini.seqexec.web.client.semanticui.elements.table.TableHeader
-import edu.gemini.seqexec.web.client.services.HtmlConstants.{iconEmpty, nbsp}
 import japgolly.scalajs.react.vdom.html_<^._
 import japgolly.scalajs.react._
 import japgolly.scalajs.react.component.Scala.Unmounted
 import japgolly.scalajs.react.extra.router.RouterCtl
-import org.scalajs.dom.html.TableRow
+import react.virtualized._
 
 import scalacss.ScalaCssReact._
 import scalaz.syntax.show._
-import scalaz.syntax.equal._
+import scalaz.syntax.traverse._
+import scalaz.syntax.std.boolean._
+import scalaz.std.AllInstances._
 
 object QueueTableBody {
-  type SequencesModel = ModelProxy[StatusAndLoadedSequencesFocus]
-
-  final case class Props(ctl: RouterCtl[SeqexecPages], sequences: SequencesModel)
-
-  // Minimum rows to display, pad with empty rows if needed
-  private val minRows = 5
-
-  def emptyRow(k: String, isLogged: Boolean): VdomTagOf[TableRow] = {
-    <.tr(
-      ^.key := k, // React requires unique keys
-      <.td(
-        ^.cls := "collapsing",
-        iconEmpty),
-      <.td(nbsp),
-      <.td(nbsp),
-      <.td(nbsp),
-      <.td(
-        SeqexecStyles.notInMobile,
-        nbsp).when(isLogged),
-      <.td(
-        SeqexecStyles.notInMobile,
-        nbsp).when(isLogged)
-    )
+  private val CssSettings = scalacss.devOrProdDefaults
+  import CssSettings._
+
+  final case class Props(ctl: RouterCtl[SeqexecPages], sequences: ModelProxy[StatusAndLoadedSequencesFocus]) {
+    private lazy val sequencesList = sequences().sequences
+    def rowGetter(i: Int): QueueRow = sequencesList.index(i).map { s =>
+        QueueRow(s.id, s.status, s.instrument, s.targetName, s.name, s.active, s.runningStep)
+      }.getOrElse(QueueRow.Zero)
+
+    def rowCount: Int =
+      sequencesList.size
+  }
+
+  // ScalaJS defined trait
+  // scalastyle:off
+  trait QueueRow extends js.Object {
+    var obsId: SequenceId
+    var status: SequenceState
+    var instrument: Instrument
+    var targetName: Option[String]
+    var name: String
+    var active: Boolean
+    var runningStep: Option[RunningStep]
+  }
+  // scalastyle:on
+  object QueueRow {
+    @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
+    def apply(obsId: SequenceId, status: SequenceState, instrument: Instrument, targetName: Option[String], name: String, active: Boolean, runningStep: Option[RunningStep]): QueueRow = {
```

Comment:
I'm just curious about this `targetName: Option[String]` in the new table... is there a reason why we'd want to show `Some target` and not the actual target name?

(Sorry... I'm having a little trouble wrapping my head around this PR.)

Reply:
the `targetName` is optional because it's obscured to non-authenticated users. For those users we remove it completely not only from the UI but also from the data model, hence the `Option`